src/dkg/round1.rs

```diff
@@ -121,11 +129,114 @@ pub fn import_secret_package(
     SerializableSecretPackage::deserialize_from(&serialized[..]).map(|pkg| pkg.into())
 }
 
+#[must_use]
+fn input_checksum<I>(min_signers: u16, signing_participants: &[I]) -> Checksum
+where
+    I: Borrow<Identity>,
+{
+    let mut hasher = SipHasher24::new();
+
+    hasher.write(&min_signers.to_le_bytes());
+
+    let mut signing_participants = signing_participants
+        .iter()
+        .map(Borrow::borrow)
+        .collect::<Vec<_>>();
+    signing_participants.sort_unstable();
+    signing_participants.dedup();
+
+    for id in signing_participants {
+        hasher.write(&id.serialize());
+    }
+
+    hasher.finish()
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct Package {
+    identity: Identity,
+    frost_package: FrostPackage,
+    group_secret_key_part: [u8; 32],
+    checksum: Checksum,
+}
+
+impl Package {
+    pub fn new(
+        identity: Identity,
+        min_signers: u16,
+        signing_participants: &[Identity],
```

Comment:
This takes a slice of `Identity` structs, but in most cases we'll want to pass a slice of references. I would suggest changing to `&[&Identity]` or, better, to `&[I] where I: Borrow<Identity>` so that multiple usage patterns can be used